### PR TITLE
(buddy) Pass JWT headers on websocket requests

### PR DESCRIPTION
**Pull request by r0mant**
Buddy PR for external contribution https://github.com/gravitational/teleport/pull/13646 that adds support for rewriting headers on websocket upgrade requests in addition to regular HTTP requests. This is mainly so we pass JWT headers.

I've also added a test for this scenario.

### CODE REVIEW

**Review by ptgott — APPROVED** [review]

**Comment by alexmv** [comment]
Thanks for taking over the PR to add tests!

I note that the two commits in #13646 got squashed into one commit in this PR, and subject of that one commit is currently "docs: Fix the reserved JWT header name."  I don't know if this project in general squashes commits, but if you do so, I'd suggest rewording the commit to headline the behavioural change, not the minor documentation tweak. :)

**Review by smallinsky — APPROVED** [review]

**Review by russjones — APPROVED** [review]
